### ISSUE TITLE
Add unique indexes to derived tables

### ISSUE THREAD

**Issue by emielver** [snowplow/dbt-snowplow-web#90]
### Describe the feature
We use the `unique_key` parameter in the config of many derived tables (see [here](https://github.com/snowplow/dbt-snowplow-web/blob/b41150684b6a3d63789178b4299a7a5c3a37cd43/models/sessions/snowplow_web_sessions.sql#L4)), but this seems to have no actual effect outside of dbt's snapshot functionality (see [here](https://docs.getdbt.com/reference/resource-configs/unique_key)). We should use indexes to actually add a unique key index to Postgres to improve performance (see [here](https://docs.getdbt.com/reference/resource-configs/postgres-configs#indexes)).

So in short, we should remove the `unique_key` parameter from the tables, and introduce `indexes` with the unique key index in our derived tables.

Discourse post for context [here](https://discourse.snowplowanalytics.com/t/snowplow-web-data-models-not-creating-primary-keys-in-postgres/7034)
